src/api/index.ts

```diff
@@ -1,27 +1,36 @@
 import {AptosClient, Types} from "aptos";
+import {isHex} from "../pages/utils";
 import {withResponseError} from "./client";
 
 export async function getTransactions(
   requestParameters: {start?: number; limit?: number},
   nodeUrl: string,
-): Promise<Types.OnChainTransaction[]> {
+): Promise<Types.Transaction[]> {
   const client = new AptosClient(nodeUrl);
+  const {start, limit} = requestParameters;
+  let bigStart;
+  if (start && 0 <= start && start <= Number.MAX_SAFE_INTEGER) {
+    bigStart = BigInt(start);
+  }
   const transactions = await withResponseError(
-    client.getTransactions(requestParameters),
+    client.getTransactions({start: bigStart, limit}),
   );
 
   // Sort in descending order
   transactions.sort((a, b) =>
-    parseInt(a.version) < parseInt(b.version) ? 1 : -1,
+    ("version" in a && parseInt(a.version)) <
```

Comment:
If the transaction doesn't have a version property, that means it's a pending transaction (and thus it's expected version will be higher than any existing versions). You could consider the version to be Infinity for this case. It would be best to extract the sorting logic to a separate function to reuse later on in this file.